### PR TITLE
Add Eval to Foundry CLI

### PR DESCRIPTION
**Pull request by KuuCi**
This PR allows users to call composer llm-foundry eval {YAML_PATH} {ARGS} while maintaining correctness with existing evaluation script. The motivation is for DLE where we want to make the CLI much more intuitive in the docker images

### CODE REVIEW

**Comment by KuuCi** [comment]
Working tests:
test-eval-rdrmLF runs:
`composer -c llmfoundry eval eval/yamls/hf_eval.yaml || (echo "Command failed - killing python" && pkill python && exit 1) `

test-eval-qQI2O7 runs:
`composer -n 8 -c llmfoundry eval eval/yamls/hf_eval.yaml || (echo "Command failed - killing python" && pkill python && exit 1) `

**Comment by b-chu** [comment]
Is there a better name for the llmfoundry/train folder?

**Review by irenedea — CHANGES_REQUESTED** [review]
I don't think we should have `llmfoundry.train.eval`. That naming doesn't make sense to me.

**Comment by b-chu** [comment]
`llmfoundry.train.train` also doesn't seem great

**Comment by KuuCi** [comment]
Test run too show working:

mcli logs test-eval-9mG0gB:
`composer -n 8 -c llmfoundry eval eval/yamls/hf_eval.yaml || (echo "Command failed - killing python" && pkill python && exit 1)`

mcli logs test-eval-VCcxoX:
`composer eval/eval.py eval/yamls/hf_eval.yaml || (echo "Command failed - killing python" && pkill python && exit 1)`

File: tests/a_scripts/eval/test_eval.py [hunk 1]
```diff
@@ -11,10 +11,10 @@
 from composer import Trainer
 from composer.loggers import InMemoryLogger
 
+from llmfoundry.command_utils import evaluate  # noqa: E402
```
Comment (dakinggg) [inline tests/a_scripts/eval/test_eval.py:14]: noqa?

File: tests/a_scripts/eval/test_eval_inputs.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from omegaconf import DictConfig
 from omegaconf import OmegaConf as om
 
-from scripts.eval.eval import main  # noqa: E402
+from llmfoundry.command_utils import evaluate  # noqa: E402
```
Comment (dakinggg) [inline tests/a_scripts/eval/test_eval_inputs.py:11]: noqa?

File: llmfoundry/command_utils/eval.py [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+from omegaconf import OmegaConf as om
+from rich.traceback import install
+
+from llmfoundry.utils import (
+    find_mosaicml_logger,
+    log_eval_analytics,
+    maybe_create_mosaicml_logger,
+)
+
+install()
```
Comment (dakinggg) [inline llmfoundry/command_utils/eval.py:25]: we removed this from train a while ago. lets remove here too

**Review by dakinggg — APPROVED** [review]

**Pull request merged**